logprep/abc/processor.py

```diff
@@ -36,11 +35,43 @@ class Config:
         tree_config: Optional[str] = field(default=None, validator=[file_validator])
         """ Path to a JSON file with a valid rule tree configuration. """
 
+    @define(kw_only=True)
+    class ProcessorMetrics(Metric):
+        """Tracks statistics about this processor"""
+
+        _prefix: str = "logprep_processor_"
+
+        number_of_processed_events: int = 0
+        """Number of events that were processed by the processor"""
+        mean_processing_time_per_event: float = 0.0
+        """Mean processing time for one event"""
+        _mean_processing_time_sample_counter: int = 0
+        number_of_warnings: int = 0
+        """Number of warnings that occurred while processing events"""
+        number_of_errors: int = 0
+        """Number of errors that occurred while processing events"""
+        generic_rule_tree: RuleTree.RuleTreeMetrics
+        """Tracker of the generic rule tree metrics"""
+        specific_rule_tree: RuleTree.RuleTreeMetrics
+        """Tracker of the specific rule tree metrics"""
+
+        def update_mean_processing_time_per_event(self, new_sample):
+            """Updates the mean processing time per event"""
+            new_avg, new_sample_counter = calculate_new_average(
+                self.mean_processing_time_per_event,
+                new_sample,
+                self._mean_processing_time_sample_counter,
+            )
+            self.mean_processing_time_per_event = new_avg
+            self._mean_processing_time_sample_counter = new_sample_counter
+
     __slots__ = [
         "name",
         "rule_class",
         "ps",
         "has_custom_tests",
+        "metrics",
+        "metric_labels",
```

Comment:
Should be declared as attribute with type hint

Reply:
done